src/main/java/org/ow2/proactive/connector/iaas/cloud/provider/jclouds/JCloudsProvider.java

```diff
@@ -129,7 +132,9 @@ public void deleteInstance(Infrastructure infrastructure, String instanceId) {
         try {
             execResponse = getComputeServiceFromInfastructure(infrastructure).runScriptOnNode(instanceId,
                                                                                               buildScriptToExecuteString(instanceScript),
-                                                                                              buildScriptOptions(instanceScript));
+                                                                                              buildScriptOptions(instanceScript,
+                                                                                                                 instanceId,
+                                                                                                                 infrastructure));
         } catch (Exception e) {
             throw new RuntimeException(e);
```

Comment:
![MINOR](https://raw.githubusercontent.com/SonarCommunity/sonar-github/master/images/severity-minor.png) Define and throw a dedicated exception instead of using a generic one. [![rule](https://raw.githubusercontent.com/SonarCommunity/sonar-github/master/images/rule.png)](http://master.ci.activeeon.com:9000/coding_rules#rule_key=squid%3AS00112)